File: 240326/처음 등장하는 위치/first-appearing-position.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        int n = sc.nextInt();
        Map<Integer, Integer> map = new TreeMap<>();
        for (int i = 0; i < n; i++) {
            int number = sc.nextInt();

            if (map.containsKey(number)) {
                continue;
            }

            map.put(number, i + 1);
        }

        for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
            System.out.printf("%d %d\n", entry.getKey(), entry.getValue());
        }
    }
}

File: 240326/처음 등장하는 위치/first-appearing-position.kt
import java.util.*

fun main() {
    val sc = Scanner(System.`in`)

    val n = sc.nextInt()

    val map: TreeMap<Int, Int> = TreeMap()
    for (i in 1..n) {
        val number = sc.nextInt()
        when {
            number !in map -> map[number] = i
            else -> {}
        }
    }

    val answer: StringBuilder = StringBuilder()
    for ((number, index) in map) {
        answer.append("$number $index\n")
    }
    print(answer)
}